### PR TITLE
Fix eval model bug [train mode to eval mode memory failed]

### PR DESCRIPTION
**Pull request by phlrain**
动态图下面 从训练和预测是要通过 model.train(), model.eval()进行切换，这个切换的时候会有严重的显存泄漏，这个pr 修复这个问题

### CODE REVIEW

**Review by JiabinYang — APPROVED** [review]
LGTM
